src/firefly/js/visualize/region/Region.js

```diff
@@ -0,0 +1,291 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+/**
+ * This Object contains the specifications of the DS9 region
+ */
+
+import Enum from 'enum';
+
+export const RegionType = new Enum(['circle', 'annulus', 'ellipse', 'ellipseannulus', 'box',
+                                    'boxannulus', 'line', 'point', 'polygon', 'text', 'undefined']);
+
+export const RegionCsys = new Enum(['PHYSICAL', 'FK4', 'B1950', 'FK5', 'J2000',
+                                    'IMAGE', 'ECLIPTIC', 'GALACTIC',
+                                    'ICRS', 'AMPLIFIER', 'LINEAR', 'DETECTOR', 'UNDEFINED']);
+
+export const RegionValueUnit = new Enum(['CONTEXT', 'DEGREE', 'RADIUS', 'ARCMIN', 'ARCSEC',
+                                         'SCREEN_PIXEL', 'IMAGE_PIXEL']);
+
+export const RegionPointType = new Enum(['circle', 'box', 'cross', 'diamond', 'x',
+                                         'arrow', 'boxcircle', 'undefined']);
+
+
+/**
+ *
+ * @param wpAry WorldPt array, items contained: line, annulus, ellipse, box: 1, line: 2, polygon: >= 3
+ * @param radiusAry RegionValue array, items contained: annulus: at least 2 radii.
+ * @param dimensionAry RegionDimension array, each item contains w, h for box and box annulus
+ *                                            or 2 radii for ellipse and ellipse annulus
+ * @param angle RegionValue
+ * @param type  RegionType
+ * @param options   RegionOptions, properties of region
+ * @param highlighted bool
+ * @constructor
+ */
+export var Region =
+    ( {wpAry = null,
+      radiusAry = null,
+      dimensionAry = null,
+      angle = null,
+      type = RegionType.undefined,
+      options = null,
+      highlighted = false}
+    ) => ({type, wpAry, radiusAry, dimensionAry, angle, options, highlighted});
```

Comment:
Let a a head and make the Factory functions lower case such a `makeRegion` instead of `Region` and `makeRegionOptions` instread of `RegionOptions`, etc